Moves around fenced div introduction and exercises (#441)
carpentries/lesson-development-training: episodes/lesson-content.md

The Carpentries Workbench, and should be left untouched.
The page content begins after the blank line that follows this header.

To get started on this home page, let's replace the template text in `index.md` with
the same short description of your lesson you added to your `README.md`.

After that, it might be a good idea to add your lesson objectives to our lesson's homepage.

::::::::::::::::::::::::::::::::::::::  challenge

## Exercise: more practice editing Markdown in GitHub

Add the objectives you defined for your lesson
as a bullet list in the `index.md` file of your lesson repository.

::::::::::::::::::::::::::::::::::::::::::::::::::

Below those, let's add the prerequisite skills you determined earlier for your lesson. 
While you can add them as a bullet point list, the Workbench provides a special formatted block 
for prerequisites, so they appear visually distinct on the page.

::: callout 

### Fenced divs

Fenced divs are structural elements within the page, which get rendered in a visually distinct 
way on the lesson website.
Fenced divs include regular Markdown content surrounded by special tags to mark 
the start and the end of such a block. 

> To mark the start of a fenced div, start a line with a start tag - at least 3 colon characters 

Reviewer: Trying to avoid repeated use of "start"

> To mark the start of a fenced div, start a line with a start tag - at least 3 colon characters 

Author: Indeed :-) - begin, start and openning to the rescue.

followed by a blank character and a keyword denoting the type of a fenced div you are 
creating (`::: fenced_div_keyword`) . 
Then you add the Markdown content of this structural block. 

> Finally, end your fenced div with an end tag consisting of at least 3 colons (`:::`).

Reviewer: ```suggestion
Finally, end your fenced div with a closing tag consisting of at least 3 colons (`:::`).
```

::::::::::::::::::::::::::::::::::::::::

## Episodes

The main body of the lesson is written in _episodes_:
the individual chunks or sections that the lesson is separated into.
The episode pages of the lesson site will be constructed from Markdown or R Markdown files
in the `episodes` folder of the lesson repository.

The `episodes` folder of the new repository contains a single example episode file,
`introduction.md`.
The content of this file includes some help with and examples of how to write Markdown files
for The Carpentries Workbench.
Like `index.md`, the episode file begins with a short header.
The fields in this header describe the episode title and the estimated time (in minutes)
required to teach it and complete the exercises.

### Creating a new episode

> Let's create a new episode file, for one of the episodes you have identified earlier in the 
> training.

Reviewer: ```suggestion
Let's create a new episode file, for one of the episodes you have identified earlier in the training.
```

First, open the "raw" view of the `introduction.md` example episode,
and copy its contents to make it easier to structure the new episode we are creating.

::: instructor

> You may ask each of the member of a the lesson team to create an episode file for

Reviewer: ```suggestion
You may ask each trainee in a lesson team to create an episode file for
```

only files with the `.md` or `.Rmd` extensions will be built into
webpages by the lesson infrastructure.

> For the content of your episode, paste the content the `introduction.md` file you copied earlier and:

Reviewer: ```suggestion
For the content of your episode, paste everything from the `introduction.md` file you copied earlier and:
```

## Adding Exercises

To create an exercise in The Carpentries Workbench,

> we use another two types of fenced divs:

Reviewer: ```suggestion
we use another two types of fenced div:
```

* `discussions`, where the main task is for participants to discuss a topic or prompt, and 
* `challenges`, where the main task is a problem to be solved.

> To start a challenge fenced div, as usual use at least 3 colons, then followed it by the 
> `challenge` keyword.

Reviewer: ```suggestion
To start a challenge fenced div, write an opening tag of at least 3 colons as usual, then follow it with the `challenge` keyword.
```

Generally speaking, you should try to transfer the drafted content from your lesson design notes into the lesson website.
Here are some suggestions for things you might try:

> 1. Add different types of exercises - e.g. add a `discussion`.

Reviewer: I like this suggestion, but do we need to add a note about `solution`s not working well with `discussion`s?

Author: Is this the issue where "solutions" do not get rendered within "discussions" @tobyhodges ? I forgot ehat the issue was but remember that something was wrong there.

Reviewer: That's right: the Workbench assumes that discussions do not have a solution.

Reviewer: ```suggestion
1. Add other exercises - e.g. add a `discussion`. Note that the lesson infrastructure does not support `solution` divs to be attached to `discussion`s.
```
